tests/end_to_end/test_vulnerability_detector/test_vulnerability_detector.py

```diff
@@ -761,27 +776,33 @@ def tests_syscollector_first_second_scan_consistency_index(self, request, host_m
         }
 
         logger.critical("Checking vulnerabilities in the second scan")
-        index_state_second_scan = get_indexer_values(host_manager, index='wazuh-states-vulnerabilities',
-                                                     greater_than_timestamp=setup_vulnerability_tests)
-        index_vulnerabilities_by_agent_second_scan = get_indexed_vulnerabilities_by_agent(index_state_second_scan)
 
-        results['vulnerabilities_index_second_scan'] = index_vulnerabilities_by_agent_second_scan
-        test_result['evidences']['vulnerabilities_index_second_scan'] = index_vulnerabilities_by_agent_second_scan
+        vuln_by_agent_index_second_scan = {}
+        for agent in host_manager.get_group_hosts('agent'):
```

Comment:
Why not follow the same format as `for agent in agents_to_check`?

Reply:
The second scan tests does not depend of the results of the first one. If one agent triggers vulnerabilities in the second one and not in the first one it will fails the consistency tests beetweeen scans instead